notebooks/Ensemble parameter exploration.ipynb

```diff
@@ -0,0 +1,1124 @@
+      "text/plain": [
+       "<Figure size 432x432 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "fig, ax = plt.subplots()\n",
+    "ax = model_results.plot(\n",
+    "    'precision_score',\n",
+    "    'recall_score',\n",
+    "    figsize=(6,6),\n",
+    "    kind='scatter',\n",
+    "    xlim=(min_both,max_both),\n",
+    "    ylim=(min_both,max_both),\n",
+    "    ax=ax)\n",
+    "\n",
+    "ax.set(xlabel='Precision score', ylabel='Recall score', \n",
+    "       title='Original model performance')\n",
+    "    \n",
+    "for k, v in model_results.iterrows():\n",
+    "    ax.text(\n",
+    "        v['precision_score'],\n",
+    "        v['recall_score'],\n",
+    "        str(v['model_name']),\n",
+    "        horizontalalignment='center',\n",
+    "        verticalalignment='top'\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "everyday-floor",
+   "metadata": {},
+   "source": [
+    "## Test ensemble models using all combinations of models"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "blind-starter",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "4083"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Get every combination of models\n",
+    "models_list = model_results.index.tolist()\n",
+    "all_combinations = []\n",
+    "for r in range(2, len(models_list) + 1):\n",
```

Comment:
why 2? I think it should be 1, so you end up with `2**12-1` models.

Reply:
If r=1 then the "combination" would just be the original model by itself - which I have the test results for anyway.